web/src/ChatBox.js

```diff
@@ -142,12 +144,36 @@ class ChatBox extends React.Component {
     Setting.showMessage("success", "Message copied successfully");
   }
 
-  likeMessage() {
-    Setting.showMessage("success", "Message liked successfully");
+  likeMessage(message) {
+    if (!message.likeUsers) {message.likeUsers = [];}
+    if (message.likeUsers.includes(this.props.account.name)) {
+      message.likeUsers = message.likeUsers.filter(user => user !== this.props.account.name);
```

Comment:
Define these in Setting.js

set = addElementToSet(set, newUser)

set = deleteElementFromSet(set, newUser)

Reply:
Use ChatGPT to write